packages/calling-stateful-client/src/CallClientState.ts

```diff
@@ -272,45 +272,64 @@ export interface RaiseHandCallFeatureState {
- * @alpha
+ * @beta
  */
-export interface TogetherModeCallFeatureState {
-  /**
-   * Proxy of {@link @azure/communication-calling#TogetherModeCallFeature.togetherModeStream}.
-   */
-  stream: TogetherModeStreamState[];
+export type CallFeatureStreamName = 'togetherMode';
+
+/* @conditional-compile-remove(together-mode) */
+/**
+ * @beta
+ */
+export interface CallFeatureStreamState extends RemoteVideoStreamState {
+  feature?: CallFeatureStreamName;
 }
 
 /* @conditional-compile-remove(together-mode) */
 /**
- * State only version of {@link @azure/communication-calling#TogetherModeVideoStream}.
- * @alpha
+ * State only version of {@link @azure/communication-calling#TogetherModeSeatingMap}.
+ * @beta
+ *
+ * Represents the seating position of a participant in Together Mode.
  */
-export interface TogetherModeStreamState {
-  /**
-   * Proxy of {@link @azure/communication-calling#TogetherModeVideoStream.id}.
-   */
-  id: number;
-  /**
-   * Proxy of {@link @azure/communication-calling#TogetherModeVideoStream.mediaStreamType}.
-   */
-  mediaStreamType: MediaStreamType;
-  /**
-   * Proxy of {@link @azure/communication-calling#TogetherModeVideoStream.isReceiving}.
-   * @public
-   */
-  isReceiving: boolean;
+export interface TogetherModeSeatingPositionState {
+  // The participant id of the participant in the seating position.
+  participantId: string;
+  // The top left offset from the top of the together mode view.
+  top: number;
+  // The left offset position from the left of the together mode view.
+  left: number;
+  //  The width of the seating area
+  width: number;
+  // The height of the seating area.
+  height: number;
+}
+
+/* @conditional-compile-remove(together-mode) */
+/**
+ * Interface representing the streams in Together Mode.
+ *
+ * @beta
+ */
+export interface TogetherModeStreamsState {
+  mainVideoStream?: CallFeatureStreamState;
+}
+
+/* @conditional-compile-remove(together-mode) */
+/**
+ * State only version of {@link @azure/communication-calling#TogetherModeCallFeature}. {@link StatefulCallClient} will
+ * automatically listen for raised hands on the call and update the state exposed by {@link StatefulCallClient} accordingly.
+ * @beta
+ */
+export interface TogetherModeCallFeatureState {
+  isActive: boolean;
   /**
-   * {@link VideoStreamRendererView} that is managed by createView/disposeView in {@link StatefulCallClient}
-   * API. This can be undefined if the stream has not yet been rendered and defined after createView creates the view.
+   * Proxy of {@link @azure/communication-calling#TogetherModeCallFeature.togetherModeStream}.
    */
-  view?: VideoStreamRendererViewState;
+  streams: TogetherModeStreamsState;
   /**
-   * Proxy of {@link @azure/communication-calling#RemoteVideoStream.size}.
+   * Proxy of {@link @azure/communication-calling#TogetherModeCallFeature.TogetherModeSeatingMap}.
    */
-  streamSize?: { width: number; height: number };
+  seatingPositions: TogetherModeSeatingPositionState[];
```

Comment:
Just curious - what was the decision behind using an array (with participantId inside the array) vs `Record<ParticipantId, SeatingPositionState>`?

Reply:
Originally, we used Map<ParticipantId, SeatingPositionState>, but there was concern since Map isn’t used in other UI APIs. The array was just a temporary placeholder to gather feedback on a better type. After reviewing the current APIs, I noticed Record was used in certain scenarios. I will update this to Record type. This should simplify the logic and eliminate unnecessary loop when accessing the seating positions